scripts/gtcadis.py

```diff
@@ -1,50 +1,68 @@
 #!/usr/bin/env python
-import argparse
-import yaml
-import io
 
+import io
+import yaml
+import argparse
 import numpy as np
-from pyne.mesh import Mesh
-from pyne.partisn import write_partisn_input, isotropic_vol_source
-from pyne.dagmc import discretize_geom, load
 from pyne import nucname
+from pyne.mesh import Mesh, IMeshTag
 from pyne.bins import pointwise_collapse
+from pyne.material import MaterialLibrary
+from pyne.partisn import write_partisn_input, isotropic_vol_source
+from pyne.dagmc import discretize_geom, load, cell_material_assignments
+from pyne.alara import calc_eta
 
 
 config_filename = 'config.yml'
 
 config = \
-    """
+"""
+# If 'True' all intermediate files created while running the script will be 
+# deleted. Change to 'False' if you want to retain all intermediate files.
+clean: True
+
 # Optional step to assess all materials in geometry for compatibility with
-# SNILB criteria
-step0:
+# SNILB criteria.
+# step 0: Information for Step 0 calculation will be read from Step 2 input
 
 # Prepare PARTISN input file for adjoint photon transport
 step1:
+    # Number of photon energy groups. This should be compatible with the dose
+    # rate conversion library. (24 or 42), default is 42.
+    p_groups: 42
     # Path to hdf5 geometry file for photon transport
-    geom_file:
+    geom_file: 
     # Volume ID of adjoint photon source cell on
     # DAGMC input [Trelis/Cubit .sat file]
-    src_cell:
+    src_cell: 
     # Volume [cm^3] of source cell (detector)
-    src_vol:
+    src_vol: 
     # Define uniformly spaced, rectagular mesh that covers entire geometry:
     # Define origin of the mesh (three entries, one per spatial dimension)
     # Define locations of the coarse meshes in each direction
     # and the number of fine meshes within corresponding coarse meshes
     # Supported: Only one entry per _mesh and _ints for a uniformly 
     # spaced mesh
     # Separate values with blank space.
-    origin:
-    xmesh:
-    xints:
-    ymesh:
-    yints:
-    zmesh:
-    zints:
+    origin: 
+    xmesh: 
+    xints: 
+    ymesh: 
+    yints: 
+    zmesh: 
+    zints: 
 
 # Calculate T matrix for each material
 step2:
+    # Path to material laden geometry (hdf5) file for adjoint neutron transport.
+    geom_file: 
```

Comment:
Perhaps there should be an input block that includes data common to all steps?

Reply:
This should be the last thing to be done I guess. When all steps are done we will have a list of all the inputs that the script needs and make the common ones generally shared by all steps.

Reply:
Or introduce the common block now, and add things once you recognize that they are common.  It will make the work more incremental